drivers/adc/ad796x/ad796x.c

```diff
@@ -0,0 +1,297 @@
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdlib.h>
+#include <errno.h>
+#include "ad796x.h"
+#include "no_os_alloc.h"
+#include "no_os_print_log.h"
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+#define AD796X_BYTES_PER_SAMPLE		4
+struct ad796x_mode_gpios {
+	uint8_t en0;
+	uint8_t en1;
+	uint8_t en2;
+	uint8_t en3;
+};
+
+static struct ad796x_mode_gpios ad796x_modes[] = {
+	[AD796X_MODE1_EXT_REF_5P0] = 		{.en3 = 1, .en2 = 0, .en1 = 0, .en0 = 1},
+	[AD796X_MODE2_INT_REF_4P0] = 		{.en3 = 1, .en2 = 0, .en1 = 0, .en0 = 1},
+	[AD796X_MODE3_EXT_REF_4P0] = 		{.en3 = 1, .en2 = 0, .en1 = 1, .en0 = 0},
+	[AD796X_MODE4_SNOOZE] = 		{.en3 = 1, .en2 = 0, .en1 = 1, .en0 = 1},
+	[AD796X_MODE5_TEST] = 			{.en3 = 0, .en2 = 1, .en1 = 0, .en0 = 0},
+	[AD796X_MODE6_INVALID] = 		{.en3 = 1, .en2 = 1, .en1 = 0, .en0 = 0},
+	[AD796X_MODE7_EXT_REF_5P0_9MHZ] = 	{.en3 = 1, .en2 = 1, .en1 = 0, .en0 = 1},
+	[AD796X_MODE8_INT_REF_4P0_9MHZ] = 	{.en3 = 1, .en2 = 1, .en1 = 0, .en0 = 1},
+	[AD796X_MODE9_EXT_REF_4P0_9MHZ] = 	{.en3 = 1, .en2 = 1, .en1 = 1, .en0 = 0},
+	[AD796X_MODE10_SNOOZE2] = 		{.en3 = 1, .en2 = 1, .en1 = 1, .en0 = 1},
+};
+
+/**
+ * @brief Read data samples from adc
+ * @param dev - The device structure.
+ * @param buf - The buffer to fill.
+ * @param samples - number of samples to read.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad796x_read_data(struct ad796x_dev *dev,
+			 uint32_t *buf,
+			 uint16_t samples)
+{
+	int ret;
+
```

Comment:
blank line can be removed, no reason to leave blank lines between declarations. see everywhere where it applies